### PR TITLE
Generate opensearch-py client from API specs

### LINKED ISSUE #489: Differences between Generated and Existing python client APIs
Take a look at issue #476  for more clarity. 

- [x] #499

Module | existing client has | generated client has
-- | -- | --
Async/__init__ | list_all_point_in_time | get_all_pits
Async/__init__ | create_point_in_time | create_pit
Async/security | health_check | health
Async/security | update_audit_config | update_audit_configuration

- [ ] 2. <b style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">APIs with Query_params differences</b><br style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">

Module | API | existing client has | generated client has | status
-- | -- | -- | -- | --
Async/__init__ | search | min_compatible_shard_node | - | min_compatible_shard_node should be removed
Async/__init__ | update_by_query | version_type | - | version_type should be removed
Async/__init__ | create_point_in_time | ignore_unavailable | allow_partial_pit_creation | ignore_unavailable,  allow_partial_pit_creation both should be present
Async/cat | cluster_manager | cluster_manager | cluster_manager_timeout, master_timeout | cluster_manager_timeout, master_timeout both should be present and not cluster_manager
Sync/cat | cluster_manager | - | master_timeout | master_timeout should be present
Async/cat.pyi | cluster_manager | - | master_timeout | master_timeout should be present
Async/cat | nodes | include_unloaded_segments | - |include_unloaded_segments should be removed
Async/cat | segments | - | cluster_manager_timeout, master_timeout | cluster_manager_timeout, master_timeout both should be present
Async/cat | plugins | include_bootstrap | - |include_bootstrap should be removed
Async/cluster | health | - | awareness_attribute, ensure_node_commissioned | awareness_attribute should be present, ensure_node_commissioned should be removed
Async/cluster | exists_component_template | cluster_manager_timeout | - | cluster_manager_timeout should be present
Async/indices | open | cluster_manager_timeout | - | cluster_manager_timeout should be present
Async/indices | delete | cluster_manager_timeout | - | cluster_manager_timeout should be present
Async/indices | exists_template | cluster_manager_timeout | - | cluster_manager_timeout should be present
Async/indices | stats | types | - | types should be removed
Async/indices | delete_data_stream | expand_wildcards | - | expand_wildcards should be removed
Async/indices | exists_index_template | cluster_manager_timeout | - | cluster_manager_timeout should be present
Async/indices | get_data_stream | expand_wildcards | - |  expand_wildcards should be removed
Async/ingest | get_pipeline | summary | - | summary should be removed
Async/nodes | stats | include_unloaded_segments | - |  include_unloaded_segments should be removed
Async/nodes | hot_threads | doc_type | - | --
Async/snapshot | get | include_repository, index_details | - | include_repository, index_details should be removed

- [ ] 3. <b style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">APIs with path_params differences(mostly related to if params are required or optional, implies whether they should have =None)</b><br style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">

Module | API | existing client has | generated client has |  status
-- | -- | -- | -- | --
Async/__init__ | create_point_in_time | index=None | index |  need to be changed in python client index!=None
Async/indices | put_mapping | index=None | index |  index=None is correct
Async/indices | create_data_stream | - | body=None | body=None is should be removed. body param is not present. 

- [x] #500

Module | API | existing client has | generated client has
-- | -- | -- | --
Async/__init__ | create | PUT | POST
Async/__init__ | put_script | PUT | POST
Async/cluster | put_component_template | PUT | POST
Async/indices | refresh | POST | GET
Async/indices | flush | POST | GET
Async/indices | clone | PUT | POST
Async/indices | put_mapping | PUT | POST
Async/indices | put_alias | PUT | POST
Async/indices | put_template | PUT | POST
Async/indices | shrink | PUT | POST
Async/indices | split | PUT | POST
Async/indices | put_index_template | PUT | POST
Async/snapshot | create | PUT | POST
Async/snapshot | create_repository | PUT | POST

- [ ] 5. <b style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">APIs that are present in one and not present in other<span class="Apple-converted-space"> </span></b><br style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">

Module | existing client has | generated client has | -
-- | -- | -- | --
Async/__init__ | terms_enum | - | terms_enum should be removed 
Async/cat | transforms | all_pit_segments, pit_segments, segment_replication | transforms  should be removed 
Async/cluster | - | delete_decommission_awareness, delete_weighted_routing, get_decommission_awareness, get_weighted_routing, put_decommission_awareness, put_weighted_routing | 
Async/indices.py | freeze, unfreeze, reload_search_analyzers, promote_data_stream, migrate_to_data_stream, disk_usage, field_usage_stats | - | freeze, unfreeze, field_usage_stats, reload_search_analyzers, disk_usage, migrate_to_data_stream  should be removed. In sync Indices client, flush_synced is working and deprecated. Async Indices.flush_synced is not present.
Async/indices.pyi | flush_synced | - | 
Async/ingest | geo_ip_stats | - | geo_ip_stats should be removed 
Async/remote_store | - | all APIs | 
Async/snapshot | repository_analyze | - | repository_analyze should be removed

- [x] #501

1. Async/__init__                   delete_point_in_time       
    
    existing:

```
@query_params()
async def delete_point_in_time(self, body=None, all=False, params=None, headers=None):
    """
    Delete a point in time
    :arg body: a point-in-time id to delete
    :arg all: set it to `True` to delete all alive point in time.
    """
    path = (
        _make_path("_search", "point_in_time", "_all")
        if all
        else _make_path("_search", "point_in_time")
    )
    return await self.transport.perform_request(
        "DELETE", path, params=params, headers=headers, body=body
    )
```

generated: 

```
@query_params()
async def delete_all_pits(self, params=None, headers=None):
    """
    Deletes all active point in time searches.
    """
    return await self.transport.perform_request(
        "DELETE", "/_search/point_in_time/_all", params=params, headers=headers
    )
    
    
@query_params()
async def delete_pit(self, body=None, params=None, headers=None):
    """
    Deletes one or more point in time searches based on the IDs passed.

    :arg body:
    """
    return await self.transport.perform_request(
        "DELETE",
        "/_search/point_in_time",
        params=params,
        headers=headers,
        body=body,
    )
```

2. Async/nodes          hot_threads                            path difference
    
    existing:

```
self.transport.perform_request(
            "GET",
            _make_path("_nodes", node_id, "hot_threads"),
            params=params,
            headers=headers,
        )

```
generated: 

```
self.transport.perform_request(
            "GET",
            _make_path("_cluster", "nodes", node_id, "hot_threads"),
            params=params,
            headers=headers,
        )
```

Tasks and dangling_indices APIs have no differences b/w existing and generated modules

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
Generate all APIs

### Issues Resolved
closes #489 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: utils/generate-api.py [hunk 1]
```diff
@@ -637,6 +642,11 @@ def read_modules():
                 paths.append({"path": key2, "methods": methods})
 
         api.update({"url": {"paths": paths}})
+        if all_paths_have_deprecation and x_deprecation_message is not None:
+            api.update({"deprecation_message": x_deprecation_message})
+
+        if namespace == "indices" and name == "put_mapping":
```
Comment (dblock) [inline utils/generate-api.py:648]: Do you want to try to extract the JSON patch logic from my other PR after this, and we can discuss whether it's a good idea, or do you want me to do it?

**Review by dblock — APPROVED** [review]
Great!

File: utils/generate-api.py (hunk 1 above)
Comment (dblock) [reply utils/generate-api.py:648]: https://github.com/opensearch-project/opensearch-py/pull/553

Comment (saimedhi) [reply utils/generate-api.py:648]: Consolidating all patches in one place within overrides is efficient. However, creating a completely new spec for those endpoints might risk becoming outdated and misaligned with the main spec.

**Pull request merged**